### PR TITLE
Fix the examples for testing java with binaries

### PR DESCRIPTION
**Pull request by byrnHDF**
The Java examples would ignore the setting of the java native command when used with a binary because the name given was a target and needed to be a file. 
Sometimes it would work when one of the default other names happened to match the native library name.

### CODE REVIEW

File: HDF5Examples/config/cmake/HDFExampleMacros.cmake [hunk 1]
```diff
@@ -227,7 +227,20 @@ macro (HDF5_SUPPORT)
         if (HDF_BUILD_JAVA AND HDF5_Java_FOUND)
           if (${HDF5_BUILD_JAVA})
             set (CMAKE_JAVA_INCLUDE_PATH "${CMAKE_JAVA_INCLUDE_PATH};${HDF5_JAVA_INCLUDE_DIRS}")
-            set (H5EX_JAVA_LIBRARY ${HDF5_JAVA_LIBRARY})
+            if (HDF5_BUILD_MODE)
+              string(TOUPPER "${HDF5_BUILD_MODE}" UPPER_BUILD_TYPE)
```
Comment (mattjala) [inline HDF5Examples/config/cmake/HDFExampleMacros.cmake:231]: Should this be `${HDF5_PACKAGE_NAME}_BUILD_MODE` to match the variable defined in hdf5-config.cmake?

File: config/cmake/hdf5-config.cmake.in [hunk 2]
```diff
@@ -29,6 +29,8 @@ set (${HDF5_PACKAGE_NAME}_VALID_COMPONENTS
     Tools
 )
 
+set (${HDF5_PACKAGE_NAME}_BUILD_MODE       @HDF_CFG_NAME@)
```
Comment (mattjala) [inline config/cmake/hdf5-config.cmake.in:32]: What's the motivation for introducing this new variable in addition to the pre-existing HDF_CFG_NAME?

File: HDF5Examples/config/cmake/HDFExampleMacros.cmake (hunk 1 above)
Comment (byrnHDF) [reply HDF5Examples/config/cmake/HDFExampleMacros.cmake:231]: That is the variable  defined in hdf5-config.cmake, which is why the update to  hdf5-config.cmake.in

File: config/cmake/hdf5-config.cmake.in (hunk 2 above)
Comment (byrnHDF) [reply config/cmake/hdf5-config.cmake.in:32]: This records the setting to supply the use of the binary.

**Review by derobins — APPROVED** [review]

**Review by mattjala — APPROVED** [review]

**Pull request merged**